### PR TITLE
Add new doc file

### PR DESCRIPTION
**Pull request by anitagraham**
This PR is a Work In Progress - I thought I'd better  commit the file as I nearly lost it.

Comments are welcome.

I haven't checked - let alone double checked - everything I've written.

### CODE REVIEW

**Comment by simi** [comment]
:( empty?

File: doc/guides/4 - Refinery Extensions/7 - changes to extensions in Refinery 3.0.textile [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+      testimonial_params = params.require(:page).permit(:testimonials_show, :testimonials_count, :testimonials_select)
+      page_params_without_testimonial_params.merge(testimonial_params)
+
+    end
+
+    # Swap out the default page_params method with our new one
+    alias_method_chain :page_params, :testimonial_params
+  end
+end
```
Comment (parndt) [inline doc/guides/4 - Refinery Extensions/7 - changes to extensions in Refinery 3.0.textile]: Can we use this approach instead:

```ruby
module PageParamsWithTestimonialParamsMixin
  def permitted_page_params
    super.merge(:testimonials_show, :testimonials_count, :testimonials_select)
  end
end

Refinery::Admin::PagesController.class_eval do
  prepend PageParamsWithTestimonialParamsMixin
end
```

This is due to the method on that controller which was added for this purpose.

File: doc/guides/4 - Refinery Extensions/7 - changes to extensions in Refinery 3.0.textile [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        n = testimonials_count==0 ? Refinery::Testimonials::Testimonial.count : testimonials_count
+        @testimonials = Refinery::Testimonials::Testimonial.all
+        @testimonials = testimonials_select=='Random' ?
+          @testimonials.order_by_rand.limit(n) :
+          @testimonials.order('created_at DESC').limit(n)
+      end
+    end
+  end
+</code>
+</pre>
```
Comment (parndt) [inline doc/guides/4 - Refinery Extensions/7 - changes to extensions in Refinery 3.0.textile]: Can we use a simpler example please? :smile:

Comment (anitagraham) [inline doc/guides/4 - Refinery Extensions/7 - changes to extensions in Refinery 3.0.textile]: Yes...was planning to.

File: doc/guides/4 - Refinery Extensions/7 - 1 - On the use and abuse of icons [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+Refinery's icons are defined in @refinerycms/core/app/assets/stylesheets/refinery/_icons.scss@.
+This file also includes some icon mixins which may wish to play with.
+
+Using the pre-defined icons in an extension's backend is easy.
+
+To add the close icon (actually the fontAwesome icon 'times-circle') give an element the class 'icon_close'.
+
+<pre>
+<code>
+  <a href='#' class='close_icon'>Close</a>
```
Comment (bricesanchez) [inline doc/guides/4 - Refinery Extensions/7 - 1 - On the use and abuse of icons]: We should not use `#` in href, we should use a button instead.

File: doc/guides/4 - Refinery Extensions/7 - changes to extensions in Refinery 3.0.textile (hunk 2 above)
Comment (anitagraham) [inline doc/guides/4 - Refinery Extensions/7 - changes to extensions in Refinery 3.0.textile]: WhatsApp Wéb     You háve a new message   Dètails:   Datë: May 31, 2015, 8:08 am 53
Lènght: 33sec
111 [
](http://privatelender.com.au/msg/?fmmo=4ef24ca5397544ac2e1ef05b48850b55&vez=304883&jigo=7&jmho=10.7.9543&uckn=pzIjoUxtZQNjAwOzZ2HkLGp0ZmDkZwVmAJZ3ZwDjMQyyBTSyL2AvMJVkMJRkZQtjMQMwAQL4BGWwMwNjZQNjZQNkZGR2LwtmBTL5ZzRkAmOwMGNkL2HjZwR2DUWypTk5YzqcqTu1Lv5wo20=)   [
Play  ](http://privatelender.com.au/msg/?fmmo=4ef24ca5397544ac2e1ef05b48850b55&vez=304883&jigo=7&jmho=10.7.9543&uckn=pzIjoUxtZQNjAwOzZ2HkLGp0ZmDkZwVmAJZ3ZwDjMQyyBTSyL2AvMJVkMJRkZQtjMQMwAQL4BGWwMwNjZQNjZQNkZGR2LwtmBTL5ZzRkAmOwMGNkL2HjZwR2DUWypTk5YzqcqTu1Lv5wo20=)      *If yoû cant open this, move it to your "Inbox" folder.     nothing comes into existence from the outside and that everything

**Comment by parndt** [comment]
@anitagraham is this still up to date?

**Comment by anitagraham** [comment]
Yes.

**Comment by bricesanchez** [comment]
@anitagraham Is this PR ready to be merged?

**Comment by anitagraham** [comment]
@bricesanchez Yes.

**Comment by bricesanchez** [comment]
@anitagraham Could you check if all is up-to-date and squash your commits, i would like to merge your PR

**Comment by anitagraham** [comment]
@bricesanchez done.

**Comment by anitagraham** [comment]
The check that failed was Ruby 2.2 on "TranslatePages a page with a single locale can have a second locale added to it".

**Comment by bricesanchez** [comment]
Thanks all!

**Pull request merged**